src/components/SignerPanel/SigningStatus.js

```diff
@@ -110,64 +107,85 @@ class SigningStatus extends React.Component {
     )
   }
   render() {
-    const { status } = this.props
+    const { theme, status } = this.props
     return (
-      <React.Fragment>
-        <Status>
+      <div>
+        <Status
+          color={theme.feedbackContent}
+          background={theme.feedbackSurface}
+        >
           <StatusImage status={status} />
-          <p>{this.getLabel()}</p>
+          <p
+            css={`
+              margin-top: ${3.5 * GU}px;
+              ${textStyle('body2')};
+            `}
+          >
+            {this.getLabel()}
+          </p>
         </Status>
-        <AdditionalInfo>{this.getInfo()}</AdditionalInfo>
+        <Info>{this.getInfo()}</Info>
         {this.getCloseButton()}
-      </React.Fragment>
+      </div>
     )
   }
 }
 
 const Status = styled.div`
-  margin-top: 80px;
-  margin-bottom: 40px;
+  background: ${({ background }) => background};
+  height: 360px;
+  margin-bottom: ${3 * GU}px;
   display: flex;
   flex-direction: column;
   justify-content: center;
   align-items: center;
-  color: ${theme.textSecondary};
-  img {
-    margin-bottom: 20px;
-  }
+  color: ${({ color }) => color};
 `
 
-const AdditionalInfo = styled(Info)`
-  p + p {
-    margin-top: 10px;
-  }
-`
+const StatusImage = ({ status }) => {
+  const theme = useTheme()
+  const color = isSignatureError(status)
+    ? theme.negative
+    : isSignatureSuccess(status)
+    ? theme.positive
+    : theme.feedbackSurfaceContentSecondary
+
+  return (
+    <div
+      css={`
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        width: 60px;
+        height: 60px;
+        border: 2px solid ${color};
+        border-radius: 50%;
+      `}
+    >
+      {isSignatureError(status) ? (
+        <IconCross
+          size="medium"
+          css={`
+            color: ${color};
+          `}
+        />
+      ) : (
+        <IconCheck
+          size="medium"
+          css={`
+            color: ${color};
+          `}
+        />
+      )}
+    </div>
+  )
+}
 
-// To skip the SVG rendering delay
-const StatusImage = ({ status }) => (
-  <StatusImageMain>
-    <StatusImageImg visible={isSigning(status)} src={imgPending} />
```

Comment:
@bpierre Not sure how much of an issue this was before (doesn't seem like a big issue right now).

Was there a large lag that made you do switch between opacity rather than conditionally rendering them?

Reply:
It was to preload the external images: no need for that anymore :+1: